### PR TITLE
Add support for arbitrary Kinds

### PR DESCRIPTION
**Pull request by jordandoig**
This pull request adds support for custom checks for arbitrary, non-controller objects. I still need to add/adjust some tests and add some pre-made checks to serve as examples, but otherwise I welcome any feedback on my progress so far.

### CODE REVIEW

File: pkg/config/schema.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ const (
 	TargetController TargetKind = "Controller"
 	// TargetIngress points to the ingress spec
 	TargetIngress TargetKind = "Ingress"
+	// TargetArbitrary points to the generic spec
+	TargetArbitrary TargetKind = "Arbitrary"
```
Comment (rbren) [inline pkg/config/schema.go:25]: Let's call this `Other` instead

**Review by rbren — COMMENTED** [review]
This is looking good!

**Review by baderbuddy — COMMENTED** [review]
I didn't get a chance to finish looking through, but I had a few comments already.

File: pkg/kube/resources.go [hunk 2]
```diff
@@ -34,8 +37,62 @@ type ResourceProvider struct {
...
 	Namespaces    []corev1.Namespace
-	Controllers   []GenericWorkload
-	Ingresses     []v1beta1.Ingress
+	Resources     resourceKindMap
+}
+
+type resourceKindMap map[string][]GenericResource
+
+func (rkm resourceKindMap) addResource(r GenericResource) {
+	if _, ok := rkm[r.Kind]; !ok {
```
[the remark this answers is not in the document]
Comment (rbren) [reply pkg/kube/resources.go]: Good call! That's pretty neat, had no idea

File: pkg/kube/resources.go [hunk 3]
```diff
@@ -34,8 +37,62 @@ type ResourceProvider struct {
...
+		rkm[r.Kind] = make([]GenericResource, 0)
+	}
+	rkm[r.Kind] = append(rkm[r.Kind], r)
+}
+
+func (rkm resourceKindMap) addResources(rs []GenericResource) {
+	for _, r := range rs {
+		rkm.addResource(r)
```
[the remark this answers is not in the document]
Comment (rbren) [reply pkg/kube/resources.go:51]: This is specifically for the multi-kind case.

**Review by makoscafee — APPROVED** [review]

**Pull request merged**